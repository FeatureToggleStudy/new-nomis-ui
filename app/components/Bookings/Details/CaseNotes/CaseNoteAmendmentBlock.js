import React from 'react'
import PropTypes from 'prop-types'
import Header from '@govuk-react/header'

import { FormattedDate, FormattedTime, FormattedDay } from '../../../intl'
import { AmendmentBlock, AmendmentText, AmendmentDate } from './CaseNoteAmendmentBlock.styles'

const CaseNoteAmendmentBlock = ({ amendment: { authorName, creationDateTime, additionalNoteText } }) => (
  <AmendmentBlock>
    <Header level={3} size="SMALL">
      Amendment
    </Header>
    <AmendmentText>{additionalNoteText}</AmendmentText>

    {creationDateTime && (
      <AmendmentDate>
        <FormattedDay value={creationDateTime} /> - <FormattedDate value={creationDateTime} /> -{' '}
        <FormattedTime value={creationDateTime} />
      </AmendmentDate>
    )}
    {authorName}
  </AmendmentBlock>
)

CaseNoteAmendmentBlock.propTypes = {
  amendment: PropTypes.shape({
    authorName: PropTypes.string.isRequired,
    creationDateTime: PropTypes.string.isRequired,
    additionalNoteText: PropTypes.string.isRequired,
  }).isRequired,
}

export default CaseNoteAmendmentBlock
