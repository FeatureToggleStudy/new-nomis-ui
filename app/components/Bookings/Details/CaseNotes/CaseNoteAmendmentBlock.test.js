import React from 'react'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import CaseNoteAmendmentBlock from './CaseNoteAmendmentBlock'

describe('<CaseNoteAmendmentBlock />', () => {
  const props = {
    amendment: {
      creationDateTime: '2019-01-29T21:15:00',
      authorName: 'User, Test',
      additionalNoteText: "Offender says he doesn't like the gym.",
    },
  }

  it('should match the default snapshot', () => {
    const tree = renderer
      .create(
        <IntlProvider locale="en">
          <CaseNoteAmendmentBlock {...props} />
        </IntlProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
