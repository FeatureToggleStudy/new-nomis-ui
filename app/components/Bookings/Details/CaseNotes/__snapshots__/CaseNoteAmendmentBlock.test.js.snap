// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseNoteAmendmentBlock /> should match the default snapshot 1`] = `
<div
  className="css-1b30ksf"
>
  <h3
    className="sc-bdVaJa dbbMZC"
  >
    Amendment
  </h3>
  <pre
    className="css-9fadba"
  >
    Offender says he doesn't like the gym.
  </pre>
  <div
    className="css-g90fw7"
  >
    <span>
      Tuesday
    </span>
     - 
    <span>
      01/29/2019
    </span>
     -
     
    <span>
      9:15 PM
    </span>
  </div>
  User, Test
</div>
`;
